src/EditorFeatures/Core/Implementation/ForegroundNotification/ForegroundNotificationService.cs

```diff
@@ -139,6 +147,15 @@ private void NotifyOnForegroundWorker()
                         {
                             // eat up cancellation
                         }
+                        catch (Exception ex) when (FatalError.ReportWithoutCrash(ex))
+                        {
+                            // The PendingWork callbacks should never throw.  In the case they do we
+                            // must ensure the IAsyncToken implementation is completed.  If it is not
+                            // then the unit test code will end up in a deadlock doing an 'await' 
+                            // on the token instance.
+                            Debug.Assert(false, ex.Message);
+                            done = true;
```

Comment:
This line seems unnecessary.  done is set to true at hte top.  It is only set to false if DoWorkFunc returns true.  But in that case, we didn't throw, and we wouldn't get here.